Start reset of global spacing with 'base' Sass partials

NicolaSaunders opened the pull request:
- Remove Normalize
- Introduce new global reset of styles
- Temporarily comment out vertical spacing styles
- Introduce logical properties

NicolaSaunders committed:
Start reset of global spacing with 'base' Sass partials

- Remove Normalize
- Introduce new global reset of styles
- Temporarily comment out vertical spacing styles
- Introduce logical properties

NicolaSaunders committed:
Logical properties for layout Sass partials

NicolaSaunders committed:
Reset of global spacing for 'core-component' Sass partials

NicolaSaunders committed:
Reset of global spacing for 'advanced-component' Sass partials

NicolaSaunders committed:
Use logical properties for accessible autocomplete styles

NicolaSaunders committed:
Reset of global spacing for 'template' Sass partials

NicolaSaunders committed:
Use logical properties for utilities styles

NicolaSaunders committed:
Comment out remaining stack includes

While we look to reset/revise vertical spacing

NicolaSaunders committed:
Update font sizes

- Add new fluid type scale using CSS custom properties (CSS variables), generated using utopia.fyi
- Replace SCSS font size mixins with CSS variables for the new font sizes

NicolaSaunders committed:
Minor adjustment to declaration order

To make it easier to remove old planet classnames when ready

NicolaSaunders committed:
Make remaining non-fluid font sizes use fluid type scale

NicolaSaunders committed:
Make remaining non-fluid font sizes use fluid type scale

NicolaSaunders committed:
Refactor line-heights

NicolaSaunders committed:
Reduce line-height on `.lead`

NicolaSaunders committed:
Start adjusting heading spacing

NicolaSaunders committed:
Reduce top spacing between adjacent headings

NicolaSaunders committed:
Unify default bottom spacing on margins

NicolaSaunders committed:
Unify paragraph spacing with space on bottom of headings

NicolaSaunders committed:
Restore offset style for link underlining

NicolaSaunders committed:
Restore list and table block margins

Consistent with paragraph block margins

NicolaSaunders committed:
Typographic adjustments

NicolaSaunders committed:
Comment out text component styles

NicolaSaunders committed:
Revise line-length style, remove text-wrap styles

NicolaSaunders committed:
Adjust list item spacing

NicolaSaunders committed:
Remove old list styles

NicolaSaunders committed:
Adjust block margin for figures

NicolaSaunders committed:
Rationalise default figcaption styles

NicolaSaunders committed:
Amend `.clean-list` class to only remove bullets

Lists in the nav element still have bullets and margin/padding removed

NicolaSaunders committed:
Add block margins to the `.component` class

No longer scoped to being inside of the main element

NicolaSaunders committed:
Set block margins on text component to zero

Treating this as an exception to the `.component` spacing rule

NicolaSaunders committed:
Add temporary bg colour to text components

NicolaSaunders committed:
Finesse component spacing

NicolaSaunders committed:
Remove default padding on `.component--columns` list

NicolaSaunders committed:
Let `.clean-list` remove all list spacing/markers, add `.nomarkers-list` for marker removal only

NicolaSaunders committed:
Adjust `.component--columns` list item spacing

NicolaSaunders committed:
Restore stack mixin on `.l-box` child of fifity-fifty component

NicolaSaunders committed:
Remove top margin on first child of fifty-fifty text content

NicolaSaunders committed:
Move translations customizations to landing template Sass partial

NicolaSaunders committed:
Remove top margin on first child of card text content

NicolaSaunders committed:
Remove top margin on first child of intro text for `.crosslinks`

NicolaSaunders committed:
Remove unwanted component spacing within `.crosslinks`

NicolaSaunders committed:
Hide auto inline margin on landing page components

NicolaSaunders committed:
Review form spacing

NicolaSaunders committed:
Allow prefooter links to be inline

NicolaSaunders committed:
Remove unneeded SVG from prefooter button

NicolaSaunders committed:
Remove top/bottom margin on first/last child of `.notes`

NicolaSaunders committed:
Merge branch 'main' into feature/spacing-and-markup

NicolaSaunders committed:
Reduce font size of top level global nav

NicolaSaunders committed:
Adjust vertical spacing inside global header banner

NicolaSaunders committed:
Adjust vertical position of global nav submenu at max-viewport

NicolaSaunders committed:
Use logical properties

NicolaSaunders committed:
Review spacing of business ecosystem and home templates

Also, tidy up video hero in design system

NicolaSaunders committed:
Review spacing of default and post templates

NicolaSaunders committed:
Review spacing of profile template

NicolaSaunders committed:
Review spacing of event template

NicolaSaunders committed:
Review spacing of event listings

NicolaSaunders committed:
Review spacing of post listings

NicolaSaunders committed:
Make CSS more readable

NicolaSaunders committed:
Condense styles for first heading in cards

Assumes use of `.card__heading` class but not necessarily first child of `.card__text`

NicolaSaunders committed:
Review spacing of people listings

NicolaSaunders committed:
Review spacing of group listings

NicolaSaunders committed:
Review spacing of Technical Reports template

NicolaSaunders committed:
Review hero spacing

NicolaSaunders committed:
Review translations spacing

NicolaSaunders committed:
Review post meta vertical spacing

NicolaSaunders committed:
Add vertical spacing to immediate sibling of `.translations`

NicolaSaunders committed:
Review spacing of "pre-footer" items

NicolaSaunders committed:
Remove component top spacing after meta section

NicolaSaunders committed:
Review pagination spacing

NicolaSaunders committed:
Less vertical space after hero on listing pages

NicolaSaunders committed:
Remove unneeded declaration

NicolaSaunders committed:
Reduce translations font size

NicolaSaunders committed:
Increase margin below hero image

NicolaSaunders committed:
Remove text component classes

Leaving div in place because it has a hook for an ID

NicolaSaunders committed:
Hide temporary colours for assessing layouts

NicolaSaunders committed:
Review margins of children of table captions

NicolaSaunders committed:
Remove excess vertical space from end of main

By removing bottom margin from last component inside main

NicolaSaunders committed:
Update example text component to remove classes

NicolaSaunders committed:
Correct typo in font size class

NicolaSaunders committed:
Update font size

NicolaSaunders committed:
Review spacing in comment thread

NicolaSaunders committed:
Add classes for largest font sizes

NicolaSaunders committed:
Remove sticky footer markup/CSS

NicolaSaunders committed:
Remove all uses of `role="list"`

No longer necessary most of the time

sean-dunwoody committed:
Change pre-footer structure

- Change pre-footer structure to accommodate paragraph content
- Modify CSS to accommodate the new HTML structure
- Cache bust CSS

NicolaSaunders committed:
Optimize TR listing page performance

Use adjacent selector to effectively lazy load individual reports using `content-visibility` and `contain-intrinsic-size`

sean-dunwoody committed:
Merge branch 'main' into feature/spacing-and-markup

sean-dunwoody committed:
Merge remote-tracking branch 'refs/remotes/origin/feature/spacing-and-markup' into feature/spacing-and-markup

sean-dunwoody committed:
Add reference to 'deploy' to help make finding this easier in the future

sean-dunwoody committed:
Correct test SSH command

sean-dunwoody committed:
Add note about SSH issues

simonrjones commented:
After talking to Sean I ran:

```
 ./vendor/bin/dep deploy staging --branch=feature/spacing-and-markup
```

But errored on the [Deployer static recipe](https://github.com/studio24/deployer-recipes/blob/main/recipe/static.php):

```
[staging]  error  in static.php on line 86:
[staging] run /usr/bin/git rev-list feature/spacing-and-markup -1
[staging] err fatal: ambiguous argument 'feature/spacing-and-markup': unknown revision or path not in the working tree.
[staging] err Use '--' to separate paths from revisions, like this:
[staging] err 'git <command> [<revision>...] -- [<file>...]'
[staging] exit code 128 (Invalid exit argument)
```

NicolaSaunders committed:
Add vertical space to `.field` following a `legend`

Adds space without unwanted extra spacing for the patterns shown when [dealing with dates and times](https://design-system.w3.org/styles/forms.html#dealing-with-dates-and-times)

NicolaSaunders committed:
Stop account menu button vertical misalignment

NicolaSaunders committed:
Restore background colour to listing page template

NicolaSaunders committed:
Remove content width limit on `<p>` in listing hero, and H1

NicolaSaunders committed:
Restore event intro vertical spacing

NicolaSaunders committed:
Allow font size classes to override default element font size

NicolaSaunders committed:
Restore bold text for in segmented options label first child

NicolaSaunders committed:
Remove style applied to `.component--text` class

Class has been removed

NicolaSaunders committed:
Update markup of pre-footer examples

NicolaSaunders committed:
Increase breakpoint for desktop nav

Addresses https://github.com/w3c/w3c-website/issues/760

NicolaSaunders committed:
Remove unnecessary SVGs from example code

NicolaSaunders committed:
Refactor form filters

Replace switcher layout with CSS Grid

NicolaSaunders committed:
Prevent empty intro links in global nav

Addresses https://github.com/w3c/w3c-website/issues/799

NicolaSaunders committed:
Reduce font sizes across typescale

NicolaSaunders committed:
Fix wrapping issue with event cards wrapping unnecessarily

Also, reduce vertical spacing on first/last paragraph

NicolaSaunders committed:
Correct font size

NicolaSaunders committed:
Reduce vertical padding on translations component

NicolaSaunders committed:
Remove style overriding vertical alignment of buttons in global nav

NicolaSaunders committed:
Remove unnecessary style

NicolaSaunders committed:
Increase global nav font size

NicolaSaunders committed:
Refactor listings and their cards

- Remove reliance on a parent class for different types of cards
- Move styles onto card variant classes
- Tighten vertical spacing, add new `.dense` class
- Add styles for `.card--user` when nested in `.card--member`

NicolaSaunders committed:
Update print styles

sean-dunwoody committed:
Update composer.lock file

sean-dunwoody committed:
Merge branch 'main' into feature/spacing-and-markup

sean-dunwoody committed:
Update composer packages

NicolaSaunders committed:
Remove leading edge padding on `.nomarkers-list`

NicolaSaunders committed:
Add `.noindent-list` class to remove leading edge padding on lists

Remove zero padding from `.nomarkers-list`
